### PR TITLE
PHOENIX-5723 PhoenixIndexImportDirectReducer cleanup method updates i…

### PR DESCRIPTION
**Pull request by tkhurana**
…ndex state to active

### CODE REVIEW

**Comment by tkhurana** [comment]
@gokceni

**Comment by swaroopak** [comment]
mind adding a test for this?

File: phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/PhoenixIndexImportDirectReducer.java [hunk 1]
```diff
@@ -100,55 +100,20 @@ private void updateCounters(IndexTool.IndexVerifyType verifyType,
 
     @Override
     protected void reduce(ImmutableBytesWritable arg0, Iterable<IntWritable> arg1,
-                          Reducer<ImmutableBytesWritable, IntWritable, NullWritable, NullWritable>.Context context)
-            throws IOException, InterruptedException
-
-    {
+            Reducer<ImmutableBytesWritable, IntWritable, NullWritable, NullWritable>.Context context)
+            throws IOException, InterruptedException {
```
Comment (gokceni) [inline phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/PhoenixIndexImportDirectReducer.java]: nit: spacing

File: phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/PhoenixIndexImportDirectReducer.java [hunk 2]
```diff
@@ -100,55 +100,20 @@ private void updateCounters(IndexTool.IndexVerifyType verifyType,
...
-            IndexToolUtil.updateIndexState(context.getConfiguration(), PIndexState.ACTIVE);
-
-            updateTasksTable(context);
+            IndexTool.IndexVerifyType verifyType = getIndexVerifyType(configuration);
+            if (verifyType != IndexTool.IndexVerifyType.NONE) {
+                updateCounters(verifyType, context);
+            }
+            IndexToolUtil.updateIndexState(configuration, PIndexState.ACTIVE);
         } catch (SQLException e) {
             LOGGER.error(" Failed to update the status to Active");
```
Comment (gokceni) [inline phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/PhoenixIndexImportDirectReducer.java]: This log statement might not be correct if the getIndexVerifyType or updateCounters throw an exception

File: phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/PhoenixIndexImportDirectReducer.java [hunk 3]
```diff
@@ -100,55 +100,20 @@ private void updateCounters(IndexTool.IndexVerifyType verifyType,
...
             throw new RuntimeException(e.getMessage());
         }
     }
-
-    private void updateTasksTable(Context context) throws SQLException, IOException {
```
Comment (gokceni) [inline phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/PhoenixIndexImportDirectReducer.java]: Please don't remove this method. Removing will cause IndexRebuildTaskIT to fail.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexExtendedIT.java [hunk 4]
```diff
@@ -314,4 +315,48 @@ public void testBuildDisabledIndex() throws Exception {
...
+            assertFalse(checkIndexState(conn, indexFullName, PIndexState.ACTIVE, 0L));
+
+            // Run the index MR job and verify that the index table rebuild fails
+            IndexToolIT.runIndexTool(true, false, schemaName, dataTableName,
+                    indexTableName, null, -1, IndexTool.IndexVerifyType.AFTER);
+
+            IndexRegionObserver.setIgnoreIndexRebuildForTesting(false);
+
+            // job failed, verify that the index table is still not in the ACTIVE state
+            assertFalse(checkIndexState(conn, indexFullName, PIndexState.ACTIVE, 0L));
```
Comment (gokceni) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexExtendedIT.java]: Is there a test to verify it goes into Active state in the sunny case?

Comment (kadirozde) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexExtendedIT.java]: @tkhurana, here just repeat the index tool run with // Run the index MR job and verify that the index table rebuild fails
            IndexToolIT.runIndexTool(true, false, schemaName, dataTableName,
                    indexTableName, null, 0, IndexTool.IndexVerifyType.AFTER); and then check the state : assertTrue(checkIndexState(conn, indexFullName, PIndexState.ACTIVE, 0L));

Comment (tkhurana) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexExtendedIT.java]: Thanks @gokceni @kadirozde I updated the test to check for Active state.

**Pull request closed without merging**